### PR TITLE
bt-48: + Create Grid Component

### LINKED ISSUE #48: Create Grid component
https://mui.com/material-ui/react-grid2/

### PR DESCRIPTION
**Pull request by JosuerBague**
image

### CODE REVIEW

**Review by mskmee — APPROVED** [review]

File: frontend/src/bundles/common/components/grid/grid.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { Grid as MUIGrid, type GridProps } from '@mui/material';
+
+const Grid: React.FC<GridProps> = ({
+    children,
+    classes,
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/grid/grid.tsx]:
```suggestion
```

File: frontend/src/bundles/common/components/grid/grid.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    md,
+    rowSpacing,
+    sm,
+    spacing,
+    sx,
+    wrap,
+    xl,
+    xs,
+    zeroMinWidth,
+    component = 'div',
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/grid/grid.tsx]:
```suggestion
    component = 'div',
    ...props
```

Comment (nikita-remeslov) [reply frontend/src/bundles/common/components/grid/grid.tsx]: we also should add className

File: frontend/src/bundles/common/components/grid/grid.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        lg={lg}
+        md={md}
+        rowSpacing={rowSpacing}
+        sm={sm}
+        spacing={spacing}
+        sx={sx}
+        wrap={wrap}
+        xl={xl}
+        xs={xs}
+        zeroMinWidth={zeroMinWidth}
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/grid/grid.tsx]:
```suggestion
        component={component}
        {...props}
```

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/components/grid/grid.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    type GridProps as MUIGridProperties,
+} from '@mui/material';
+
+type Properties = MUIGridProperties;
+
+const Grid: React.FC<Properties> = ({
+    children,
+    className = '',
+    component = 'div',
+    ...properties
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/grid/grid.tsx]:
```suggestion
    ...props
```

File: frontend/src/bundles/common/components/grid/grid.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+}) => (
+    <MUIGrid className={className} component={component} {...properties}>
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/grid/grid.tsx]:
```suggestion
    <MUIGrid className={className} component={component} {...props}>
```

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/components/grid/grid.tsx (hunk 4 above)
Comment (JosuerBague) [reply frontend/src/bundles/common/components/grid/grid.tsx]: Hello, I tried props first but linter wouldn't allow it, hence properties.

Comment (nikita-remeslov) [reply frontend/src/bundles/common/components/grid/grid.tsx]: merge development branch

**Comment by nikita-remeslov** [comment]
@JoshDagat Also you should import you component to the root component.ts file

**Comment by JosuerBague** [comment]
> @JoshDagat Also you should import you component to the root component.ts file

Done.

File: frontend/src/bundles/common/components/grid/grid.tsx (hunk 4 above)
Comment (JosuerBague) [reply frontend/src/bundles/common/components/grid/grid.tsx]: Development branch merged and changes made (properties -> props).

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**